crates/noirc_evaluator/src/ssa/acir_gen/mod.rs

```diff
@@ -564,33 +577,50 @@ impl Context {
                                 call_stack,
                             });
                         }
-                        let result_type =
-                            dfg.type_of_value(dfg.instruction_results(instruction)[0]);
-                        let value = self.create_default_value(&result_type)?;
+                    } else {
+                        let value = match store_value {
+                            Some(store_value) => {
+                                let store_value = self.convert_value(store_value, dfg);
+                                AcirValue::Array(array.update(index, store_value))
+                            }
+                            None => array[index].clone(),
+                        };
+
                         self.define_result(dfg, instruction, value);
                         return Ok(());
                     }
-
-                    let value = match store_value {
-                        Some(store_value) => {
-                            let store_value = self.convert_value(store_value, dfg);
-                            AcirValue::Array(array.update(index, store_value))
-                        }
-                        None => array[index].clone(),
-                    };
-
-                    self.define_result(dfg, instruction, value);
-                    return Ok(());
                 }
             }
             AcirValue::DynamicArray(_) => (),
         }
         let resolved_array = dfg.resolve(array);
         let map_array = last_array_uses.get(&resolved_array) == Some(&instruction);
         if let Some(store) = store_value {
-            self.array_set(instruction, array, index, store, dfg, map_array)?;
+            // In case of side effects:
+            // - we replace the index by predicate*index
+            // - we replace the value by predicate*value + (1-predicate)*dummy, where dummy is the value of the array at the requested index
```

Comment:
```suggestion
            // - We replace the value by predicate*store_value + (1-predicate)*dummy, where dummy is the value of the array at the requested index. It is a dummy value because in the case of a false predicate, the value stored at the requested index will be itself.
```
How do we feel about expanding this comment like so?

Reply:
Done